src/details/ArborX_DetailsUtils.hpp

```diff
@@ -119,6 +119,10 @@ create_layout_right_mirror_view_and_copy(ExecutionSpace const &execution_space,
   {
     return src;
   }
+  else if constexpr (has_compatible_layout)
+  {
+    return Kokkos::create_mirror_view_and_copy(memory_space, src);
+  }
```

Comment:
Can't you do
```C++
  if constexpr (has_compatible_layout)
  {
    return Kokkos::create_mirror_view_and_copy(memory_space, src);
  }
```
? If the memory space is the same `create_mirror_view_and_copy` also just returns a (shallow) copy.

Reply:
Good point. Wonder why we didn't do it.